src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/correlation/extractordata/ExtractorData.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jmeter.protocol.http.correlation.extractordata;
```

Comment:
Same note here regarding base class and hierarchy

Reply:
This class already follows inheritance. It has child classes like BoundaryExtractorData. I would request you to please verify.

Reply:
This class already follows inheritance. It has child classes like BoundaryExtractorData, RegexExtractorData, etc.
 Please share your comments if you think our understanding for your suggestion is incorrect.